### PR TITLE
WT-11391 Create 2 new stats for eviction

### PR DESCRIPTION
**Pull request by adelinexchen**

### CODE REVIEW

**Comment by adelinexchen** [comment]
@quchenhao I am unsure about how to name or provide descriptions to these two stats, or whether or not I need to add more in other spots.

File: dist/stat_data.py [hunk 1]
```diff
@@ -939,6 +939,7 @@ def __init__(self, name, desc, flags=''):
     ##########################################
     # Cache and eviction statistics
     ##########################################
+
```
Comment (quchenhao) [inline dist/stat_data.py:942]: Remove empty line

File: dist/stat_data.py [hunk 2]
```diff
@@ -952,6 +953,7 @@ def __init__(self, name, desc, flags=''):
     CacheStat('cache_eviction_blocked_no_ts_checkpoint_race_2', 'eviction gave up due to detecting a tombstone without a timestamp ahead of the selected on disk update'),
     CacheStat('cache_eviction_blocked_no_ts_checkpoint_race_3', 'eviction gave up due to detecting a tombstone without a timestamp ahead of the selected on disk update after validating the update chain'),
     CacheStat('cache_eviction_blocked_no_ts_checkpoint_race_4', 'eviction gave up due to detecting update chain entries without timestamps after the selected on disk update'),
+    CacheStat('cache_eviction_blocked_no_ts_checkpoint_race_5', 'eviction gave up due to no progress being made'),
```
Comment (quchenhao) [inline dist/stat_data.py:956]: We should remove the name of no ts as this one is not caused by an update wihtout a timestamp. I would name it cache_eviction_blocked_no_progress

File: src/reconcile/rec_write.c [hunk 3]
```diff
@@ -2040,7 +2041,8 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
      * unless we're the checkpoint thread. Big pages take a lot of writes, avoid wasting work.
      */
     if (!last_block && WT_BTREE_SYNCING(btree) && !WT_SESSION_BTREE_SYNC(session))
-        return (__wt_set_return(session, EBUSY));
+        WT_STAT_CONN_DATA_INCR(session, cache_multi_block_reconciliation);
```
Comment (quchenhao) [inline src/reconcile/rec_write.c:2044]: This is wrong. You should add a {} block and move the return statement inside.

File: dist/stat_data.py [hunk 4]
```diff
@@ -998,6 +1000,7 @@ def __init__(self, name, desc, flags=''):
     CacheStat('cache_hs_write_squash', 'history store table writes requiring squashed modifies'),
     CacheStat('cache_inmem_split', 'in-memory page splits'),
     CacheStat('cache_inmem_splittable', 'in-memory page passed criteria to be split'),
+    CacheStat('cache_multi_block_reconciliation', 'multiple block reconciliation whilst checkpoint is running'),
```
Comment (quchenhao) [inline dist/stat_data.py:1003]: I would name it cache_eviction_blocked_multi_block_reconcilation_during_checkpoint

Comment (adelinexchen) [reply dist/stat_data.py]: fixed https://github.com/wiredtiger/wiredtiger/pull/10046/commits/a8303dcd3f551be75085fcc46e7412e18e18c8fe

File: dist/stat_data.py [hunk 5]
```diff
@@ -998,6 +999,7 @@ def __init__(self, name, desc, flags=''):
     CacheStat('cache_hs_write_squash', 'history store table writes requiring squashed modifies'),
     CacheStat('cache_inmem_split', 'in-memory page splits'),
     CacheStat('cache_inmem_splittable', 'in-memory page passed criteria to be split'),
+    CacheStat('cache_eviction_blocked_multi_block_reconcilation_during_checkpoint', 'multiple block reconciliation whilst checkpoint is running'),
```
Comment (quchenhao) [inline dist/stat_data.py:1002]: multi-block reconciliation blocked whilst checkpoint is running

Comment (adelinexchen) [reply dist/stat_data.py]: fixed https://github.com/wiredtiger/wiredtiger/pull/10046/commits/c11ee27e165685f470eb81ca1e7890b66b305db7

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by Sean04 — APPROVED** [review]
LGTM

**Comment by adelinexchen** [comment]
evergreen merge

**Pull request merged**